src/main/java/io/supertokens/session/Session.java

```diff
@@ -669,59 +671,144 @@ public static String[] revokeSessionUsingSessionHandles(Main main,
+            return new String[0];
+        } else {
+            String revokedSessionHandles[] = new String[numberOfSessionsRevoked];
+            int currentIndex = 0;
+            for (String sessionHandle : sessionHandles) {
+                if (!validHandles.contains(sessionHandle)) {
+                    continue; // no need to check if the sessionHandle was invalid in the first place
+                }
+                if (tenantIdentifierWithStorage.getSessionStorage()
+                        .getSession(tenantIdentifierWithStorage, sessionHandle) == null) {
+                    revokedSessionHandles[currentIndex] = sessionHandle;
+                    currentIndex++;
+                }
+            }
+            return revokedSessionHandles;
+        }
     }
 
     @TestOnly
-    public static String[] revokeAllSessionsForUser(Main main,
-                                                    String userId) throws StorageQueryException {
+    public static String[] revokeAllSessionsForUser(Main main, String userId) throws StorageQueryException {
         Storage storage = StorageLayer.getStorage(main);
-        return revokeAllSessionsForUser(
-                new TenantIdentifierWithStorage(null, null, null, storage), userId);
+        return revokeAllSessionsForUser(main,
+                new AppIdentifierWithStorage(null, null, storage), userId);
     }
 
-    public static String[] revokeAllSessionsForUser(TenantIdentifierWithStorage tenantIdentifierWithStorage,
+    public static String[] revokeAllSessionsForUser(Main main, AppIdentifierWithStorage appIdentifierWithStorage,
+                                                    String userId) throws StorageQueryException {
+        String[] sessionHandles = getAllNonExpiredSessionHandlesForUser(main, appIdentifierWithStorage, userId);
+        return revokeSessionUsingSessionHandles(main, appIdentifierWithStorage, sessionHandles);
+    }
+
+    public static String[] revokeAllSessionsForUser(Main main, TenantIdentifierWithStorage tenantIdentifierWithStorage,
                                                     String userId) throws StorageQueryException {
         String[] sessionHandles = getAllNonExpiredSessionHandlesForUser(tenantIdentifierWithStorage, userId);
-        return revokeSessionUsingSessionHandles(tenantIdentifierWithStorage, sessionHandles);
+        return revokeSessionUsingSessionHandles(main, tenantIdentifierWithStorage.toAppIdentifierWithStorage(), sessionHandles);
     }
 
     @TestOnly
     public static String[] getAllNonExpiredSessionHandlesForUser(Main main, String userId)
             throws StorageQueryException {
         Storage storage = StorageLayer.getStorage(main);
-        return getAllNonExpiredSessionHandlesForUser(
-                new TenantIdentifierWithStorage(null, null, null, storage), userId);
+        return getAllNonExpiredSessionHandlesForUser(main,
+                new AppIdentifierWithStorage(null, null, storage), userId);
+    }
+
```

Comment:
add tests for non multi tenancy use (basically these funcitons being called when multi tenancy is not enabled)